pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -314,11 +314,45 @@ func (c *Cluster) resolveNode(nodeName, deviceClass string) *cephv1.Node {
 	if rookNode == nil {
 		return nil
 	}
-	rookNode.Resources = k8sutil.MergeResourceRequirements(rookNode.Resources, cephv1.GetOSDResources(c.spec.Resources, deviceClass))
 
+	rookNode.Resources = c.getOSDResources(rookNode, rookNode.ResourceClassName, deviceClass)
 	return rookNode
 }
 
+func (c *Cluster) resourceRequirementFromResourceClass(resourceClassName string) (corev1.ResourceRequirements, error) {
+	for _, i := range c.ValidStorage.ResourceClasses {
+		if i.Name == resourceClassName {
+			return i.Resources, nil
+		}
+	}
+	return corev1.ResourceRequirements{}, errors.Errorf("failed to find resourceClassName %s", resourceClassName)
+}
+
+func resolveDevice(devices []cephv1.Device, deviceClass string) string {
```

Comment:
Can I also get resourceName without iteration?
Little confused about how we set/get this property from[ config-map](https://github.com/rook/rook/blob/bb7306e6ee109775ed70e7b4239b5e93c22a8a33/pkg/operator/ceph/cluster/osd/status.go#L313).

Reply:
Why are you searching all devices in the list for any of them that set the device class? Seems like we would need to match specifically to the class requested for a device, so I wouldn't expect to iterate all the devices like this.

Reply:
@travisn, Yes we need to find the ResourceClass of a device, 
I would have only one more way in my mind, that we should create a map like `map[nodeName]map[deviceClassName]deviceResourceClassName` and update all these values in one iteration.
 
or I think it will be convenient if we think in the perspective of time complexity because I think we can have only 3 different forms of the disk/deviceClass  (hdd, nvme, ssd), so it will be a quick iteration.